File: iPhone/Habibi/PhraseDetailsViewController.m
//
//  PhraseDetailsViewController.m
//  Habibi
//
//  Created by B2BConnect on 19/07/14.
//
//

#import "PhraseDetailsViewController.h"

@interface PhraseDetailsViewController ()

@end

@implementation PhraseDetailsViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view.
    [self.view addSubview:[APPDEL createCustomNavView:NO doneBtn:NO]];

   

}

-(void)viewWillAppear:(BOOL)animated
{
    
    
    if([self.catName isEqualToString:@"Mood"] || [self.catName isEqualToString:@"Responses"])
    {
        self.btnFemaleSound.hidden = YES;
        self.btnMaleSound.hidden = YES;
        toGenderLabel.hidden = YES;
    }
    
    
    
    
   // NSLog(@"%s",__FUNCTION__);
    int toGender;
    
    if ([[[NSUserDefaults standardUserDefaults]objectForKey:@"habibi_Gender"] isEqualToString:@"Male"])
    {
        toGender=1;
        [self PlaySoundToMale:self.btnMaleSound];
    }
    else
    {
        toGender=2;
        [self playSoundToFemale:self.btnFemaleSound];
    }
    
    return;

    NSMutableArray *arr= [self fetchConvertedPhraseData:self.habibiPhrase_id];
    if (arr.count>0)
    {
        lblPhrase1.text=self.strHabibiPhrase;
        lblPhraseTranslation1.text=[NSString stringWithFormat:@"   %@",[[arr objectAtIndex:0]objectForKey:@"native_phrase"]];
        lblPhraseTranslation2.text=[NSString stringWithFormat:@"   %@",[[arr objectAtIndex:0]objectForKey:@"phonetic_spelling"]];
        lblPhraseTranslation3.text=[NSString stringWithFormat:@"   %@",[[arr objectAtIndex:0]objectForKey:@"proper_phonetic_spelling"]];
    }
    else
    {
       NSMutableArray *arr= [self fetchPharseData:self.habibiPhrase_id toGender:toGender fromGender:toGender];
        if (arr.count>0)
        {
            lblPhrase1.text=self.strHabibiPhrase;
            lblPhraseTranslation1.text=[NSString stringWithFormat:@"   %@",[[arr objectAtIndex:0]objectForKey:@"native_phrase"]];
            lblPhraseTranslation2.text=[NSString stringWithFormat:@"   %@",[[arr objectAtIndex:0]objectForKey:@"phonetic_spelling"]];
            lblPhraseTranslation3.text=[NSString stringWithFormat:@"   %@",[[arr objectAtIndex:0]objectForKey:@"proper_phonetic_spelling"]];
        }

    }
    
}
- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
-(NSMutableArray *)fetchConvertedPhraseData:(NSNumber *)phraseID
{
    int fromGender;
    int toGender;
    if ([[[NSUserDefaults standardUserDefaults]objectForKey:@"habibi_Gender"] isEqualToString:@"Male"])
    {
        toGender=1;
    }
    else
    {
        toGender=2;
    }
    
    if ([[[NSUserDefaults standardUserDefaults]objectForKey:@"my_Gender"] isEqualToString:@"Male"])
    {
        fromGender=1;
    }
    else
    {
        fromGender=2;
    }
    
    
            FMDatabase *database=[FMDatabase databaseWithPath:[APPDEL dbPath]];
        [database open];
        NSMutableArray *arrPhraseDetailsList=[[NSMutableArray alloc]init];
        FMResultSet *results = [database executeQuery:[NSString stringWithFormat:@"SELECT * FROM phrase where habibi_phrase_id  = %d and from_gender=%d and to_gender=%d",phraseID.intValue,fromGender,toGender]];
    NSLog(@"Qry :-- %@",[NSString stringWithFormat:@"SELECT * FROM phrase where habibi_phrase_id  = %d and from_gender=%d and to_gender=%d",phraseID.intValue,fromGender,toGender]);
        while([results next]) {
            int catID  = [results intForColumn:@"_id"];
            int habibi_phrase_id  = [results intForColumn:@"habibi_phrase_id"];
            int language  = [results intForColumn:@"language"];
            int dialect  = [results intForColumn:@"dialect"];
            int from_gender  = [results intForColumn:@"from_gender"];
            int to_gender  = [results intForColumn:@"to_gender"];
            NSString *native_phrase = [results stringForColumn:@"native_phrase"];
            NSString *phonetic_spelling = [results stringForColumn:@"phonetic_spelling"];
            NSString *proper_phonetic_spelling = [results stringForColumn:@"proper_phonetic_spelling"];
            
            
            
            NSDictionary *dict=[NSDictionary dictionaryWithObjectsAndKeys:
                                [NSNumber numberWithInt:catID],@"id",
                                [NSNumber numberWithInt:habibi_phrase_id],@"habibi_phrase_id",
                                [NSNumber numberWithInt:language],@"language",
                                [NSNumber numberWithInt:dialect],@"dialect",
                                [NSNumber numberWithInt:from_gender],@"from_gender",
                                [NSNumber numberWithInt:to_gender],@"to_gender",
                                native_phrase,@"native_phrase",
                                phonetic_spelling,@"phonetic_spelling",
                                proper_phonetic_spelling,@"proper_phonetic_spelling",nil];
            
            [arrPhraseDetailsList addObject:dict];
            
        }
        [database close];
   
    return arrPhraseDetailsList;
}
- (IBAction)copyBtnOnAction:(id)sender
{
    UIButton *btn=sender;
    UIPasteboard *pb = [UIPasteboard generalPasteboard];
    UIAlertView *alert=[[UIAlertView alloc]initWithTitle:@"Message" message:nil delegate:nil cancelButtonTitle:@"OK" otherButtonTitles:nil, nil];

    switch (btn.tag) {
        case 1:
        {
            [pb setString:lblPhraseTranslation1.text];
            alert.message=[NSString stringWithFormat:@"Text has been copied to your clipboard"];
            break;
        }
        case 2:
        {
            [pb setString:lblPhraseTranslation2.text];
            alert.message=[NSString stringWithFormat:@"%@ has been copied to your clipboard",[pb string]];

            break;
        }
        case 3:
        {
            [pb setString:lblPhraseTranslation3.text];
            alert.message=[NSString stringWithFormat:@"%@ has been copied to your clipboard",[pb string]];

            break;
        }
    }
    
    [alert show];
}
/*
- (IBAction)playSoundOnClick:(id)sender
{
    NSURL *fileURL;
    NSCharacterSet *notAllowedChars = [[NSCharacterSet characterSetWithCharactersInString:@"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ "] invertedSet];
    NSString *resultString = [[self.strHabibiPhrase componentsSeparatedByCharactersInSet:notAllowedChars] componentsJoinedByString:@""];
    resultString=[resultString stringByReplacingOccurrencesOfString:@" " withString:@"_"];
    
    
    NSString *strGender;
    if ([[[NSUserDefaults standardUserDefaults]objectForKey:@"habibi_Gender"] isEqualToString:@"Male"] && [[[NSUserDefaults standardUserDefaults]objectForKey:@"my_Gender"] isEqualToString:@"Male"])
    {
        strGender=@"_m_m";
    }
    else if ([[[NSUserDefaults standardUserDefaults]objectForKey:@"habibi_Gender"] isEqualToString:@"Female"] && [[[NSUserDefaults standardUserDefaults]objectForKey:@"my_Gender"] isEqualToString:@"Female"])
    {
        strGender=@"_f_f";
    }
    
    else if ([[[NSUserDefaults standardUserDefaults]objectForKey:@"habibi_Gender"] isEqualToString:@"Male"] && [[[NSUserDefaults standardUserDefaults]objectForKey:@"my_Gender"] isEqualToString:@"Female"])
    {
        strGender=@"_f_m";
        
    }
    else if ([[[NSUserDefaults standardUserDefaults]objectForKey:@"habibi_Gender"] isEqualToString:@"Female"] && [[[NSUserDefaults standardUserDefaults]objectForKey:@"my_Gender"] isEqualToString:@"Male"])
    {
        strGender=@"_m_f";
        
    }
    
    resultString= [resultString stringByAppendingString:strGender];
    NSString *resultStr=[resultString lowercaseString];
    NSLog (@"Result: %@", resultStr);
    
    NSString* filePath = [[NSBundle mainBundle] pathForResource:resultString
                                                         ofType:@"m4a"];
    
    
    NSFileManager *fileManager = [NSFileManager defaultManager];
    
    
    if ([fileManager fileExistsAtPath:filePath])
    {
        NSLog(@"Exists");
    }
    else
    {
        NSString *str= [resultStr stringByReplacingCharactersInRange:NSMakeRange(resultStr.length-2,2) withString:[resultStr substringWithRange:NSMakeRange(resultStr.length-4,2)]];
        NSLog (@"Result: %@", str);
        
        filePath = [[NSBundle mainBundle] pathForResource:str ofType:@"m4a"];
        
        if ([fileManager fileExistsAtPath:filePath])
        {
            NSLog(@"Exists");
        }
        else
        {
            NSString *str= [resultStr stringByReplacingCharactersInRange:NSMakeRange(resultStr.length-2,2) withString:@""];
            NSLog (@"Result: %@", str);
            
            
            filePath = [[NSBundle mainBundle] pathForResource:[str lowercaseString] ofType:@"m4a"];
            
            if ([fileManager fileExistsAtPath:filePath])
            {
                NSLog(@"Exists");
            }
            else
            {
                NSString *str1= [str stringByReplacingCharactersInRange:NSMakeRange(str.length-2,2) withString:@""];
                NSLog (@"Result: %@", str1);
                
                filePath = [[NSBundle mainBundle] pathForResource:str1 ofType:@"m4a"];
                
                if ([fileManager fileExistsAtPath:filePath])
                {
                    NSLog(@"Exists");
                }
                else
                {
                    
                    NSLog (@"sound file not available");
                    UIAlertView *alert=[[UIAlertView alloc]initWithTitle:@"Message" message:@"Sound file not available" delegate:nil cancelButtonTitle:@"OK" otherButtonTitles:nil, nil];
                    [alert show];
                    alert=nil;
                    
                }
                
            }
            
            
        }
        
    }
    
    // NSString *soundFilePath = [[NSBundle mainBundle] pathForResource:@"no" ofType: @"m4a"];
    if ([fileManager fileExistsAtPath:filePath])
    {
        fileURL = [[NSURL alloc] initFileURLWithPath:filePath];
        
        NSError *error;
        
        self.player = [[AVAudioPlayer alloc] initWithContentsOfURL:fileURL error:&error];
        [self.player play];
    }
}

*/

// Old backup

- (IBAction)playSoundOnClick:(id)sender
{
    NSURL *fileURL;
    NSCharacterSet *notAllowedChars = [[NSCharacterSet characterSetWithCharactersInString:@"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ "] invertedSet];
    NSString *resultString = [[self.strHabibiPhrase componentsSeparatedByCharactersInSet:notAllowedChars] componentsJoinedByString:@""];
    resultString=[resultString stringByReplacingOccurrencesOfString:@" " withString:@"_"];
    
    
    NSString *fromString;
    
    if([[[NSUserDefaults standardUserDefaults]objectForKey:@"my_Gender"] isEqualToString:@"Male"])
    {
        fromString = @"_m";
    }
    else
    {
        fromString = @"_f";
    }
    
    NSString *toString;
    if(self.btnMaleSound.isSelected)
    {
        toString = @"_m";
    }
    else
    {
        toString = @"_f";

    }

    
    
    NSString *strGender = [NSString stringWithFormat:@"%@%@",fromString,toString];
    
    /*
    if ([[[NSUserDefaults standardUserDefaults]objectForKey:@"habibi_Gender"] isEqualToString:@"Male"] && [[[NSUserDefaults standardUserDefaults]objectForKey:@"my_Gender"] isEqualToString:@"Male"])
    {
        strGender=@"_m_m";
    }
    else if ([[[NSUserDefaults standardUserDefaults]objectForKey:@"habibi_Gender"] isEqualToString:@"Female"] && [[[NSUserDefaults standardUserDefaults]objectForKey:@"my_Gender"] isEqualToString:@"Female"])
    {
        strGender=@"_f_f";
    }
    
    else if ([[[NSUserDefaults standardUserDefaults]objectForKey:@"habibi_Gender"] isEqualToString:@"Male"] && [[[NSUserDefaults standardUserDefaults]objectForKey:@"my_Gender"] isEqualToString:@"Female"])
    {
        strGender=@"_f_m";

    }
    else if ([[[NSUserDefaults standardUserDefaults]objectForKey:@"habibi_Gender"] isEqualToString:@"Female"] && [[[NSUserDefaults standardUserDefaults]objectForKey:@"my_Gender"] isEqualToString:@"Male"])
    {
        strGender=@"_m_f";

    }
     */
    
   resultString= [resultString stringByAppendingString:strGender];
    NSString *resultStr=[resultString lowercaseString];
    NSLog (@"Result: %@", resultStr);

    NSString* filePath = [[NSBundle mainBundle] pathForResource:resultString
                                                         ofType:@"m4a"];
    
    
    NSFileManager *fileManager = [NSFileManager defaultManager];
    
    
    if ([fileManager fileExistsAtPath:filePath])
    {
        NSLog(@"Exists");
    }
    else
    {
       NSString *str= [resultStr stringByReplacingCharactersInRange:NSMakeRange(resultStr.length-4,2) withString:[resultStr substringFromIndex:resultString.length-2]];
        NSLog (@"Result: %@", str);

        filePath = [[NSBundle mainBundle] pathForResource:str ofType:@"m4a"];
    
        if ([fileManager fileExistsAtPath:filePath])
        {
            NSLog(@"Exists");
        }
        else
        {
            NSString *str= [resultStr stringByReplacingCharactersInRange:NSMakeRange(resultStr.length-4,2) withString:@""];
            NSLog (@"Result: %@", str);
            
            
            filePath = [[NSBundle mainBundle] pathForResource:[str lowercaseString] ofType:@"m4a"];
            
            if ([fileManager fileExistsAtPath:filePath])
            {
                NSLog(@"Exists");
            }
            else
            {
                NSString *str1= [str stringByReplacingCharactersInRange:NSMakeRange(str.length-2,2) withString:@""];
                NSLog (@"Result: %@", str1);
                
                filePath = [[NSBundle mainBundle] pathForResource:str1 ofType:@"m4a"];
                
                if ([fileManager fileExistsAtPath:filePath])
                {
                    NSLog(@"Exists");
                }
                else
                {
                    
                    NSLog (@"sound file not available");
                    UIAlertView *alert=[[UIAlertView alloc]initWithTitle:@"Message" message:@"Sound file not available" delegate:nil cancelButtonTitle:@"OK" otherButtonTitles:nil, nil];
                    [alert show];
                    alert=nil;
                    
                }
                
            }
            
            
        }
    
    }
    
   // NSString *soundFilePath = [[NSBundle mainBundle] pathForResource:@"no" ofType: @"m4a"];
    if ([fileManager fileExistsAtPath:filePath])
    {
        fileURL = [[NSURL alloc] initFileURLWithPath:filePath];

        NSError *error;
        
         self.player = [[AVAudioPlayer alloc] initWithContentsOfURL:fileURL error:&error];
        [self.player play];
    }
}
 
 

- (IBAction)PlaySoundToMale:(id)sender
{
    UIButton *btn=sender;
    int toGender;
    
    if (btn.isSelected) {
        //[btn setSelected:NO];
       // [btn setBackgroundColor:[UIColor clearColor]];
       // [self.btnFemaleSound setSelected:YES];
       // [self.btnFemaleSound setBackgroundColor:[UIColor blueColor]];
         //toGender=2;
    }
    else
    {
        [btn setSelected:YES];
        [btn setBackgroundColor:[UIColor blueColor]];
        [self.btnFemaleSound setSelected:NO];
        [self.btnFemaleSound setBackgroundColor:[UIColor clearColor]];
        
        toGender=1;
    
    
    NSMutableArray *arr= [self fetchPharseData:self.habibiPhrase_id toGender:toGender fromGender:toGender];
    if (arr.count>0)
    {
        lblPhrase1.text=self.strHabibiPhrase;
        lblPhraseTranslation1.text=[NSString stringWithFormat:@"   %@",[[arr objectAtIndex:0]objectForKey:@"native_phrase"]];
        lblPhraseTranslation2.text=[NSString stringWithFormat:@"   %@",[[arr objectAtIndex:0]objectForKey:@"phonetic_spelling"]];
        lblPhraseTranslation3.text=[NSString stringWithFormat:@"   %@",[[arr objectAtIndex:0]objectForKey:@"proper_phonetic_spelling"]];
    }
    }
    
    [self fixBorders];

}

- (IBAction)playSoundToFemale:(id)sender
{
    UIButton *btn=sender;
    int toGender;
    if ([btn isSelected])
    {
//        [btn setSelected:NO];
//        [btn setBackgroundColor:[UIColor clearColor]];
//        [self.btnMaleSound setSelected:YES];
//        [self.btnMaleSound setBackgroundColor:[UIColor blueColor]];
//         toGender=1;
    }
    else
    {
        [btn setSelected:YES];
        [btn setBackgroundColor:[UIColor blueColor]];
        [self.btnMaleSound setSelected:NO];
        [self.btnMaleSound setBackgroundColor:[UIColor clearColor]];
           toGender=2;

  
    
    NSMutableArray *arr= [self fetchPharseData:self.habibiPhrase_id toGender:toGender fromGender:toGender];
    if (arr.count>0)
    {
        lblPhrase1.text=self.strHabibiPhrase;
        lblPhraseTranslation1.text=[NSString stringWithFormat:@"   %@",[[arr objectAtIndex:0]objectForKey:@"native_phrase"]];
        lblPhraseTranslation2.text=[NSString stringWithFormat:@"   %@",[[arr objectAtIndex:0]objectForKey:@"phonetic_spelling"]];
        lblPhraseTranslation3.text=[NSString stringWithFormat:@"   %@",[[arr objectAtIndex:0]objectForKey:@"proper_phonetic_spelling"]];
    }
    }
    
    [self fixBorders];
}

-(NSMutableArray *)fetchPharseData:(NSNumber *)phraseID toGender:(int)toGender fromGender:(int)fromGender
{

    FMDatabase *database=[FMDatabase databaseWithPath:[APPDEL dbPath]];
    [database open];
    NSMutableArray *arrPhraseDetailsList=[[NSMutableArray alloc]init];
    FMResultSet *results = [database executeQuery:[NSString stringWithFormat:@"SELECT * FROM phrase where habibi_phrase_id  = %d and from_gender=%d and to_gender=%d",phraseID.intValue,fromGender,toGender]];
    NSLog(@"Qry :-- %@",[NSString stringWithFormat:@"SELECT * FROM phrase where habibi_phrase_id  = %d and from_gender=%d and to_gender=%d",phraseID.intValue,fromGender,toGender]);
    while([results next]) {
        int catID  = [results intForColumn:@"_id"];
        int habibi_phrase_id  = [results intForColumn:@"habibi_phrase_id"];
        int language  = [results intForColumn:@"language"];
        int dialect  = [results intForColumn:@"dialect"];
        int from_gender  = [results intForColumn:@"from_gender"];
        int to_gender  = [results intForColumn:@"to_gender"];
        NSString *native_phrase = [results stringForColumn:@"native_phrase"];
        NSString *phonetic_spelling = [results stringForColumn:@"phonetic_spelling"];
        NSString *proper_phonetic_spelling = [results stringForColumn:@"proper_phonetic_spelling"];
        
        
        
        NSDictionary *dict=[NSDictionary dictionaryWithObjectsAndKeys:
                            [NSNumber numberWithInt:catID],@"id",
                            [NSNumber numberWithInt:habibi_phrase_id],@"habibi_phrase_id",
                            [NSNumber numberWithInt:language],@"language",
                            [NSNumber numberWithInt:dialect],@"dialect",
                            [NSNumber numberWithInt:from_gender],@"from_gender",
                            [NSNumber numberWithInt:to_gender],@"to_gender",
                            native_phrase,@"native_phrase",
                            phonetic_spelling,@"phonetic_spelling",
                            proper_phonetic_spelling,@"proper_phonetic_spelling",nil];
        
        [arrPhraseDetailsList addObject:dict];
        
    }
    [database close];
    
    return arrPhraseDetailsList;
}

-(void)fixBorders
{
    for (UIView *vw in self.view.subviews) {
        if([vw isKindOfClass:[UIButton class]])
        {
            if(vw.backgroundColor == [UIColor blueColor])
            {
                vw.layer.cornerRadius = 2.0;
                vw.layer.borderColor = [UIColor whiteColor].CGColor;
                vw.layer.borderWidth = 2.0;
                vw.clipsToBounds = YES;
            }
            else
            {
                vw.layer.cornerRadius = 2.0;
                vw.layer.borderColor = [UIColor whiteColor].CGColor;
                vw.layer.borderWidth = 0.0;
                vw.clipsToBounds = YES;
            }
            
        }
    }
    
}


@end

File: iPhoneSwift/Habibi/PhraseDetailsViewController.swift
//
//  PhraseDetailsViewController.swift
//  Habibi
//
//  Created by Baljeet Singh on 16/03/15.
//  Copyright (c) 2015 B2BConnect. All rights reserved.
//

import UIKit
import AVFoundation

class PhraseDetailsViewController: UIViewController {

    @IBOutlet weak var lblPhrase1: UILabel!
    
    @IBOutlet weak var btnPhraseTranslation1: UIButton!
    
    @IBOutlet weak var btnPhraseTranslation2: UIButton!
    
    @IBOutlet weak var btnPhraseTranslation3: UIButton!
    
    @IBOutlet weak var btnFemaleSound: UIButton!
    
    @IBOutlet weak var btnMaleSound: UIButton!
    
   
    @IBOutlet weak var toGenderLabel: UILabel!
    
    var habibiPhrase_id : NSString!
    
    var strHabibiPhrase : NSString!
    
    var catName : NSString!
    
    var appdel = AppDelegate()
    
    var player : AVAudioPlayer = AVAudioPlayer()

    @IBAction func PlaySoundToMale(sender: UIButton)
    {
        let btn = sender
        var toGender :Int
        
        if (btn.selected) {
            
        }
        else
        {
            
            btn.selected = true
            btn.backgroundColor = UIColor.blueColor()
            self.btnFemaleSound.selected = false
            self.btnFemaleSound.backgroundColor = UIColor.clearColor()
            
            toGender = 1
            var arr : NSMutableArray = ModelManager.instance .fetchPharseData(phraseID: self.habibiPhrase_id, toGender: NSString(format: "%d", toGender), fromGender: NSString(format: "%d", toGender))
            
            NSLog("final arr :- %@", arr.description)
            
            if arr.count > 0
            {
                lblPhrase1.text=self.strHabibiPhrase
                
                var dict : NSDictionary = arr.objectAtIndex(0) as NSDictionary
                var nativePhrase : NSString = dict .objectForKey("native_phrase") as NSString
                var phonetic_spelling : NSString = dict .objectForKey("phonetic_spelling") as NSString
                var proper_phonetic_spelling : NSString = dict.objectForKey("proper_phonetic_spelling") as NSString
                
                btnPhraseTranslation1 .setTitle(nativePhrase, forState: UIControlState.Normal)
                btnPhraseTranslation2 .setTitle(phonetic_spelling, forState: UIControlState.Normal)
                btnPhraseTranslation3 .setTitle(proper_phonetic_spelling, forState: UIControlState.Normal)
                
            }
            
        }
        
        
        self.fixBorders()
    }
    
    @IBAction func playSoundToFemale(sender: UIButton)
    {
        let btn = sender
        var toGender :Int
        
        if (btn.selected) {
            
        }
        else
        {

            
            btn.selected = true
            btn.backgroundColor = UIColor.blueColor()
            self.btnMaleSound.selected = false
            self.btnMaleSound.backgroundColor = UIColor.clearColor()
            toGender = 2
            
            
            var arr : NSMutableArray =  ModelManager.instance.fetchPharseData(phraseID: self.habibiPhrase_id, toGender: NSString(format: "%d", toGender), fromGender: NSString(format: "%d", toGender))
            NSLog("final arr :- %@", arr.description)

            if arr.count > 0
            {
                lblPhrase1.text = self.strHabibiPhrase
                
                var dict : NSDictionary = arr.objectAtIndex(0) as NSDictionary
                var nativePhrase : NSString = dict .objectForKey("native_phrase") as NSString
                var phonetic_spelling : NSString = dict .objectForKey("phonetic_spelling") as NSString
                var proper_phonetic_spelling : NSString = dict.objectForKey("proper_phonetic_spelling") as NSString
                
                btnPhraseTranslation1 .setTitle(nativePhrase, forState: UIControlState.Normal)
                btnPhraseTranslation2 .setTitle(phonetic_spelling, forState: UIControlState.Normal)
                btnPhraseTranslation3 .setTitle(proper_phonetic_spelling, forState: UIControlState.Normal)
                
            }
        }
        
        self.fixBorders()
    }
    
    @IBAction func playSoundOnClick(sender: UIButton)
    {
//        NSURL *fileURL;
//        NSCharacterSet *notAllowedChars = [[NSCharacterSet characterSetWithCharactersInString:@"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ "] invertedSet];
//        NSString *resultString = [[self.strHabibiPhrase componentsSeparatedByCharactersInSet:notAllowedChars] componentsJoinedByString:@""];
//        resultString=[resultString stringByReplacingOccurrencesOfString:@" " withString:@"_"];
//        
//        
//        NSString *fromString;
        
        
        var fileURL : NSURL
        var notAllowedChars : NSCharacterSet = NSCharacterSet (charactersInString: "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ ")
        var result : NSArray = self.strHabibiPhrase.componentsSeparatedByCharactersInSet(notAllowedChars)
        
        var resultString : NSString = result .componentsJoinedByString("")
        
        resultString = resultString .stringByReplacingOccurrencesOfString(" ", withString: "_")
        
        var fromString : NSString
        
        
        let my_Gender : NSString =   NSUserDefaults .standardUserDefaults() .objectForKey("my_Gender") as NSString
        
        
        
        
//        if([[[NSUserDefaults standardUserDefaults]objectForKey:@"my_Gender"] isEqualToString:@"Male"])
        if(my_Gender == "Male")
        {
            fromString = "_m";
        }
        else
        {
            fromString = "_f";
        }
        
      //  NSString *toString;
        var toString : NSString
        
        if self.btnMaleSound.selected
        {
            toString = "_m";
        }
        else
        {
            toString = "_f";
            
        }
        
        
        
        //NSString *strGender = [NSString stringWithFormat:@"%@%@",fromString,toString];
        
        var strGender : NSString = NSString(format: "%@%@", fromString,toString)
        
        
        //resultString= [resultString stringByAppendingString:strGender];
        
        var resultStr : NSString = resultString .stringByAppendingString(strGender).lowercaseString
        
//        NSString *resultStr=[resultString lowercaseString];
//        NSLog (@"Result: %@", resultStr);
        println(resultStr)
        
//        NSString* filePath = [[NSBundle mainBundle] pathForResource:resultString
//        ofType:@"m4a"];
        
        var filePath : NSString = NSBundle.mainBundle().pathForResource(resultString, ofType: "m4a")!
        
       // NSFileManager *fileManager = [NSFileManager defaultManager];
    
        
        let fileManager =  NSFileManager.defaultManager()
        
        
//        if ([fileManager fileExistsAtPath:filePath])
         if fileManager .fileExistsAtPath(filePath)
        {
            println("exists")
            
        }
        else
        {
         //   NSString *str= [resultStr stringByReplacingCharactersInRange:NSMakeRange(resultStr.length-4,2) withString:[resultStr substringFromIndex:resultString.length-2]];
            
            var str : NSString = resultStr .stringByReplacingCharactersInRange(NSMakeRange(resultStr.length-4, 2), withString:resultStr.substringFromIndex(resultString.length-2))
            
            NSLog("Result %@", str)
            
           // NSLog (@"Result: %@", str);
            
          //  filePath = [[NSBundle mainBundle] pathForResource:str ofType:@"m4a"];
            filePath = NSBundle.mainBundle().pathForResource(str, ofType: "m4a")!
            
            
//            if ([fileManager fileExistsAtPath:filePath])
            if fileManager .fileExistsAtPath(filePath)
            {
                println("result")
            }
            else
            {
//                NSString *str= [resultStr stringByReplacingCharactersInRange:NSMakeRange(resultStr.length-4,2) withString:@""];
//                NSLog (@"Result: %@", str);
                
                var str : NSString = resultStr .stringByReplacingCharactersInRange(NSMakeRange(resultStr.length-4, 2), withString: "")
                NSLog("result :- %@", str)
                
                
                //filePath = [[NSBundle mainBundle] pathForResource:[str lowercaseString] ofType:@"m4a"];
                
                filePath = NSBundle.mainBundle().pathForResource(str.lowercaseString, ofType: "m4a")!
                
                
                if fileManager.fileExistsAtPath(filePath)
                {
                    println("Exists")
                }
                else
                {
//                    NSString *str1= [str stringByReplacingCharactersInRange:NSMakeRange(str.length-2,2) withString:@""];
                    var str1 : NSString = str .stringByReplacingCharactersInRange(NSMakeRange(str.length-2, 2), withString: "")
                    
//                    NSLog (@"Result: %@", str1);
                    NSLog("result :- %@", str1)
                    
                   // filePath = [[NSBundle mainBundle] pathForResource:str1 ofType:@"m4a"];
                    
                    filePath = NSBundle .mainBundle().pathForResource(str1, ofType: "m4a")!
                    
                    if (fileManager.fileExistsAtPath(filePath))
                    {
                             println("exists")
                    }
                    else
                    {
                        
                        println("sound file not available");
//                        UIAlertView *alert=[[UIAlertView alloc]initWithTitle:@"Message" message:@"Sound file not available" delegate:nil cancelButtonTitle:@"OK" otherButtonTitles:nil, nil];
//                        [alert show];
//                        alert=nil;
                        
                        var alert : UIAlertView = UIAlertView(title: "Message", message: "Sound file not available", delegate: nil, cancelButtonTitle: "OK", otherButtonTitles: "", "")
                        alert.show()
                        
                        
                    }
                    
                }
                
                
            }
            
        }
        
        if (fileManager.fileExistsAtPath(filePath))
        {
            //fileURL = [[NSURL alloc] initFileURLWithPath:filePath];
            fileURL = NSURL(fileURLWithPath: filePath)!
            
//            NSError *error;
            var error : NSError = NSError()
            
//            self.player = [[AVAudioPlayer alloc] initWithContentsOfURL:fileURL error:&error];
            self.player = AVAudioPlayer(contentsOfURL: fileURL, error: nil)
            self.player.play()
            
        }
    }
    override func viewDidLoad() {
        super.viewDidLoad()
        appdel = UIApplication.sharedApplication().delegate as AppDelegate
        
        self.view .addSubview(appdel.createCustomNavView(false, doneVisible: false))

        // Do any additional setup after loading the view.
    }

    override func viewWillAppear(animated: Bool)
    {

         let cat_name = self.catName
        
        if cat_name == "Mood" || cat_name == "Responses"
        {
            self.btnFemaleSound.hidden = true
            self.btnMaleSound.hidden = true
            self.toGenderLabel.hidden = true
        }
        
        let habibiGender =  NSUserDefaults .standardUserDefaults() .objectForKey("habibi_Gender")as NSString
        
        
        
        if habibiGender == "Male"
        {
            self.PlaySoundToMale(btnMaleSound)
        }
        else
        {
            self.playSoundToFemale(btnFemaleSound)
        }
        

    }
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */
    
    func fixBorders()
    {
        
        for o : AnyObject in self.view.subviews
        {
            // println("Hello World")
            
            if let v = o as? UIButton
            {
                //println("Print btn")
                //  println(v)
                if (v.backgroundColor == UIColor.blueColor())
                {
                    v.layer.cornerRadius = 2.0
                    v.layer.borderColor = UIColor.whiteColor().CGColor
                    v.layer.borderWidth = 2.0
                    v.clipsToBounds = true
                    
                }
                else
                {
                    v.layer.cornerRadius = 2.0
                    v.layer.borderColor = UIColor.whiteColor().CGColor
                    v.layer.borderWidth = 0.0
                    v.clipsToBounds = true
                }
                
            }
            
        }
        
    }

}
